src/app/rundown-view/components/minishelf/mini-shelf.component.html

```diff
@@ -0,0 +1,20 @@
+<div
+  class="c-mini-shelf"
+  title="{{ segment.name }}"
+  (click)="executeAction()"
+>
+<!--  TODO: add ThumbnailComponent and use it here-->
+    <img
+        class="c-mini-shelf__thumbnail"
+        src="{{ mediaPreviewUrl }}"
+        alt="{{ segment.name? segment.name : 'Title is not available.' }}"
+        (error)=handleMissingImage($event)
+    />
+    <div class="c-mini-shelf__note">
+        <span class="c-mini-shelf__title"
+        >{{ getSanitizedTitle() ? getSanitizedTitle() : 'Title is not available.'  | slice:0:30 | uppercase }}{{ segment.name.length > 30 ? '...' : '' }}</span>
```

Comment:
The newlines for the span tag seem a bit odd here. And the the clipping of the segment name, can be done with a `max-width: <some-number>%;` and `text-overflow: ellipsis;` on the `.c-mini-shelf__title` class. That way the browser determines when to clip the segment name based on available space.